packages/atomic/src/components/search/atomic-rating-range-facet/atomic-rating-range-facet.spec.ts

```diff
@@ -0,0 +1,545 @@
+        element.shadowRoot?.querySelector('[part="facet"]');
+      expect(facetContainer).toBeNull();
+    });
+  });
+
+  describe('collapsed state', () => {
+    it('should render values when not collapsed', async () => {
+      const {element} = await renderRatingRangeFacet({
+        props: {isCollapsed: false},
+      });
+      const values = element.shadowRoot?.querySelector('[part="values"]');
+      expect(values).toBeDefined();
+    });
+
+    it('should not render values when collapsed', async () => {
+      const {element} = await renderRatingRangeFacet({
+        props: {isCollapsed: true},
+      });
+      const values = element.shadowRoot?.querySelector('[part="values"]');
+      expect(values).toBeNull();
+    });
+  });
+
+  describe('rating display', () => {
+    it('should display rating icons for each value', async () => {
+      const {element} = await renderRatingRangeFacet();
+      const ratingIcons = element.shadowRoot?.querySelectorAll(
+        '[part~="value-rating-icon"]'
+      );
+      expect(ratingIcons!.length).toBeGreaterThan(0);
+    });
+
+    it('should use custom icon when provided', async () => {
+      const customIcon = '<svg>custom</svg>';
+      const {element} = await renderRatingRangeFacet({
+        props: {icon: customIcon},
+      });
+      expect(element.icon).toBe(customIcon);
+    });
+  });
+
+  describe('interaction', () => {
+    it('should call toggleSingleSelect when clicking link value', async () => {
+      mockedNumericFacet = buildFakeNumericFacet({
+        implementation: {
+          toggleSingleSelect: vi.fn(),
+        },
+      });
+
+      const {locators} = await renderRatingRangeFacet();
+      const firstValue = locators.getFacetValueByPosition(0);
+      await firstValue.click();
+
+      expect(mockedNumericFacet.toggleSingleSelect).toHaveBeenCalled();
+    });
+
+    it('should call deselectAll when clicking clear button', async () => {
+      mockedNumericFacet = buildFakeNumericFacet({
+        implementation: {
+          deselectAll: vi.fn(),
+        },
+        state: {
+          values: [
+            {
+              start: 4,
+              end: 5,
+              endInclusive: false,
+              state: 'selected',
+              numberOfResults: 10,
+            },
+          ],
+        },
+      });
+
+      const {locators} = await renderRatingRangeFacet();
+      await locators.clearButton.click();
+
+      expect(mockedNumericFacet.deselectAll).toHaveBeenCalled();
+    });
+  });
+
+  describe('disconnectedCallback', () => {
+    it('should stop watching dependencies when disconnected', async () => {
+      mockedFacetConditionsManager = buildFakeFacetConditionsManager({});
+      const stopWatchingSpy = vi.spyOn(
+        mockedFacetConditionsManager,
+        'stopWatching'
+      );
+
+      const {element} = await renderRatingRangeFacet();
+      element.disconnectedCallback();
+
+      expect(stopWatchingSpy).toHaveBeenCalled();
+    });
+  });
```

Comment:
Nit: this would be closer to the pattern we use in atomic-facet (using element.remove instead of calling the disconnectedCallback directly).

```suggestion
  describe('when removed from the DOM (#disconnectedCallback)', () => {
    it('should stop watching dependencies', async () => {
      const stopWatchingSpy = vi.spyOn(
        mockedFacetConditionsManager,
        'stopWatching'
      );

      const {element} = await renderRatingRangeFacet();
      element.disconnectedCallback();

      expect(stopWatchingSpy).toHaveBeenCalled();
    });
  });
```